client/my-sites/post-type-list/index.jsx

```diff
@@ -122,6 +132,18 @@ class PostTypeList extends Component {
 		return scrollContainer.scrollTop;
 	}
 
+	hasListFullyLoaded() {
+		const { lastPage, isRequestingPosts } = this.props;
+		const { maxRequestedPage } = this.state;
+
+		return ! isRequestingPosts && maxRequestedPage >= lastPage;
+	}
+
+	isTruncatedResults() {
+		const { siteId, lastPage } = this.props;
+		return null === siteId && this.hasListFullyLoaded() && lastPage > MAX_ALL_SITES_PAGES;
+	}
```

Comment:
Let's pull the contents of this method into `renderMaxPagesNotice` - it's only used once, and I'm not crazy about the name anyway.

Reply:
It will probably also be useful to pull out some logic to get the maximum page we should actually request (`siteId === null ? MAX_ALL_SITES_PAGES : lastPage`).  Then, we need to make sure we never actually request more than this number of pages, and show the max pages notice after loading the 10th page.